test/connection_switching_test.rb

```diff
@@ -585,31 +592,35 @@ class ShardedModel < ActiveRecord::Base
     end
   end
 
-  describe "alternative connections" do
-    it "not interfere with other connections" do
-      assert_using_database('ars_test', Account)
-      assert_using_database('ars_test', Ticket)
-      assert_using_database('ars_test_alternative', Email)
+  if ActiveRecord::VERSION::MAJOR < 5
 
-      ActiveRecord::Base.on_shard(0) do
+    describe "alternative connections" do
```

Comment:
I removed it (see PR #76) and rebased this branch on top of that removal branch.